model/CSVDialog.cc

```diff
@@ -34,6 +53,111 @@ void CSVDialog::reportFromFile(const std::string& input, const std::string& outp
+        
+        // Verify the remote server's certificate
+        ctx.set_verify_mode(ssl::verify_peer | ssl::context::verify_fail_if_no_peer_cert);    
+        ctx.set_default_verify_paths();
+        
+        // tag::ctx_setup_source[]
+        boost::certify::enable_native_https_server_verification(ctx);
+        // end::ctx_setup_source[]        
+        		        
+        // These objects perform our I/O
+        tcp::resolver resolver{ioc};
+        ssl::stream<tcp::socket> stream{ioc, ctx};        
+        
+        // tag::stream_setup_source[]
+        boost::certify::set_server_hostname(stream, string(what[2].first, what[2].second));
+        boost::certify::sni_hostname(stream, host);
+        // end::stream_setup_source[]         
+
+        // Look up the domain name
+        auto const results = resolver.resolve(host, what[1]);
+                
+        // Make the connection on the IP address we get from a lookup
+        boost::asio::connect(stream.next_layer(), results.begin(), results.end());                   
+             
+        // Perform the SSL handshake
+        stream.handshake(ssl::stream_base::client);             
+
+        // Set up an HTTP GET request message
+        http::request<http::string_body> req{http::verb::get, target, version};
+        req.set(http::field::host, host);
+        req.set(http::field::user_agent, BOOST_BEAST_VERSION_STRING);
+
+        // Send the HTTP request to the remote host
+        http::write(stream, req);
+
+        // This buffer is used for reading and must be persisted
+        boost::beast::flat_buffer buffer;
+
+        // Declare a container to hold the response
+        http::response_parser<http::string_body> res;
+        res.body_limit((std::numeric_limits<std::uint64_t>::max)());
+
+        // Receive the HTTP response
+        http::read(stream, buffer, res);
+                                                       
+        // Dump the outstream into a temporary file for loading it into Minsky' CSV parser 
+        boost::filesystem::path temp = boost::filesystem::unique_path();
+        const std::string tempStr    = temp.native();
+                
+        std::ofstream outFile(tempStr, std::ofstream::out);  
+        outFile << res.get().body();                                            
+             
+        // Gracefully close the socket
+        boost::system::error_code ec;
+        stream.shutdown(ec);
+        if (ec == boost::asio::error::eof)
+        {
+            // Rationale:
+            // http://stackoverflow.com/questions/25587403/boost-asio-ssl-async-shutdown-always-finishes-with-an-error
+            ec.assign(0, ec.category());
+        }
+        if (ec)
+            throw boost::system::system_error{ec}; 
+            
+        // Return the file name for loading the in csvimport.tcl 
+        return tempStr;
+
+        // If we get here then the connection is closed gracefully
+    }
+    catch(std::exception const& e)
```

Comment:
This try/catch can probably be eliminated. It looks like beast does not throw, so the only exception will be the one you're throwing on line 147, plus the one you will throw on checking the response code on line 128.

Reply:
Try/catch gone and added a throw to test the response code.